### ISSUE TITLE
Does not detect `start-storybook` script

### ISSUE THREAD

**Issue by IanVS** [joshbolduc/vscode-story-explorer#460]
I've got a normal, standard storybook project, with `start-storybook` in my `node_modules/.bin` but I get the error:

```
Launching the Storybook server failed. The path to the Storybook server script couldn't be determined. If this project uses Storybook, you may need to install your project's dependencies, or specify a path manually in settings.
```

Here's a simple repo that demonstrates this issue: https://github.com/IanVS/storybook-mdx2-reproduction/tree/fix-mdx2

If I add `${workspaceFolder}/node_modules/.bin/start-storybook` to `storyExplorer.server.internal.storybookBinaryPath`, then it launches, but it seems like that should be the default.

**Comment by joshbolduc** [comment]
Thanks for the report. I tried reproing this and #461 using that repo and branch, but everything started up and loaded normally for me.

Could you share some more info about the environment where you're running VSCode/the extension (OS, version, etc.)? Wonder if this might be platform-specific.

**Comment by IanVS** [comment]
Hmmmm, I'm using 

story-explorer 0.5.6,
macOS 12.3.1, 
npm 8.3.0 & 8.12.1, 
vscode 1.68.1, 
node 16.14.2,

I don't have chrome installed, only firefox, safari, and brave. (Open Story in Browser also does nothing).

I disabled all other extensions except this one, and still have the same behavior.  Let me know if there's anything else you can think of I can try checking.  Thanks!

**Comment by joshbolduc** [comment]
Hmm, interesting. Do you have the `files.exclude` setting configured to ignore `node_modules` by chance?

<img width="893" alt="image" src="https://user-images.githubusercontent.com/25474557/176822308-59c2dd44-314e-4c15-a98a-7a96310387b2.png">

**Comment by joshbolduc** [comment]
In case it's the issue above, I think this test build here would resolve it: https://github.com/joshbolduc/vscode-story-explorer/suites/7173193353/artifacts/285911369

**Comment by IanVS** [comment]
Hmmm, nope, I haven't changed that setting. 
<img width="222" alt="image" src="https://user-images.githubusercontent.com/4616705/176906280-2c43ffc2-8bfc-4125-8912-5c699b610961.png">


Here are all my user settings:

```json
{
  "[go]": {
    "editor.defaultFormatter": "golang.go"
  },
  "[javascript]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode",
    "editor.formatOnSave": true
  },
  "[json]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "[php]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode",
    "editor.formatOnSave": false
  },
  "[typescript]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode",
    "editor.formatOnSave": true
  },
  "[typescriptreact]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode",
    "editor.formatOnSave": true
  },
  "auto-close-tag.SublimeText3Mode": true,
  "auto-close-tag.activationOnLanguage": [
    "xml",
    "php",
    "blade",
    "ejs",
    "jinja",
    "javascript",
    "javascriptreact",
    "typescript",
    "typescriptreact",
    "plaintext",
    "markdown",
    "vue",
    "liquid",
    "erb",
    "lang-cfml",
    "cfml",
    "HTML (Eex)",
    "django-html",
    "HTML",
    "php",
    "mdx"
  ],
  "auto-close-tag.excludedTags": [
    "area",
    "base",
    "br",
    "col",
    "command",
    "embed",
    "hr",
    "img",
    "input",
    "keygen",
    "link",
    "meta",
    "param",
    "source",
    "track",
    "wbr",
    "OwnProps",
    "boolean",
    "string",
    "number"
  ],
  "cSpell.userWords": ["OIDC", "semibold", "signout", "TOTP", "viewports"],
  "css.validate": false,
  "editor.accessibilitySupport": "off",
  "editor.cursorWidth": 2,
  "editor.defaultFormatter": "esbenp.prettier-vscode",
  "editor.find.addExtraSpaceOnTop": false,
  "editor.fontFamily": "Hack, Input Sans Narrow, Menlo, Monaco, 'Courier New', monospace",
  "editor.formatOnSave": true,
  "editor.hover.above": false,
  "editor.minimap.renderCharacters": false,
  "editor.multiCursorModifier": "alt",
  "editor.renderWhitespace": "all",
  "editor.suggestSelection": "first",
  "editor.tabSize": 2,
  "editor.unicodeHighlight.allowedCharacters": {
    "’": true
  },
  "editor.wordWrap": "on",
  "editorconfig.generateAuto": false,
  "eslint.onIgnoredFiles": "warn",
  "eslint.validate": [
    "javascript",
    "javascriptreact",
    "typescript",
    "typescriptreact"
  ],
  "explorer.confirmDragAndDrop": false,
  "files.insertFinalNewline": true,
  "git.autofetch": true,
  "git.ignoreLegacyWarning": true,
  "gitlens.codeLens.enabled": false,
  "gitlens.statusBar.enabled": false,
  "javascript.updateImportsOnFileMove.enabled": "always",
  "liveServer.settings.root": "/storybook-static",
  "openRelatedFiles.createFileMap": {
    ".jsx": [".module.css", ".stories.js", ".test.js"],
    ".tsx": [".module.css", ".stories.tsx"]
  },
  "prettier.requireConfig": true,
  "redhat.telemetry.enabled": false,
  "search.exclude": {
    "**/node_cache": true
  },
  "search.followSymlinks": false,
  "storyExplorer.logLevel": "debug",
  "storyExplorer.server.internal.storybookBinaryPath": "${workspaceFolder}/node_modules/.bin/start-storybook",
  "stylelint.validate": [
    "css",
    "html",
    "less",
    "markdown",
    "sass",
    "scss",
    "source.css.styled",
    "source.markdown.math",
    "styled-css",
    "sugarss",
    "svelte",
    "vue",
    "vue-html",
    "vue-postcss",
    "xml",
    "xsl"
  ],
  "svgPreview.style": {
    "html": {
      "background-image": "linear-gradient(45deg, #141414 25%, transparent 25%, transparent 75%, #141414 75%, #141414), linear-gradient(45deg, #141414 25%, transparent 25%, transparent 75%, #141414 75%, #141414)",
      "background-position": "0 0, 13px 13px",
      "background-size": "26px 26px"
    }
  },
  "task.slowProviderWarning": ["typescript"],
  "terminal.integrated.defaultProfile.osx": "zsh",
  "terminal.integrated.lineHeight": 1.2,
  "typescript.inlayHints.functionLikeReturnTypes.enabled": true,
  "typescript.inlayHints.parameterNames.enabled": "literals",
  "typescript.inlayHints.parameterTypes.enabled": true,
  "typescript.inlayHints.variableTypes.enabled": true,
  "typescript.preferences.importModuleSpecifier": "non-relative",
  "typescript.updateImportsOnFileMove.enabled": "always",
  "workbench.colorCustomizations": {
    "editorCursor.foreground": "#23bee8",
    "terminalCursor.foreground": "#23bee8"
  },
  "workbench.editor.limit.value": 6,
  "workbench.editorAssociations": {
    "*.ipynb": "jupyter.notebook.ipynb"
  },
  "workbench.list.horizontalScrolling": true,
  "workbench.startupEditor": "none",
  "workbench.statusBar.visible": true,
  "zentabs.maximumOpenedTabs": 8
}
```

If I remove `storyExplorer.server.internal.storybookBinaryPath`, I get the failure.

**Comment by joshbolduc** [comment]
I've published 0.5.8 with some extra logging to help diagnose both issues. If you have a chance to repro with debug logging enabled and can share the logs, that would be helpful!

**Comment by IanVS** [comment]
Here's what I see with the new debug output:

```
[debug] Detected 0 matches for start-storybook
[debug] Webview host received message {"type":0}
[error] Failed to complete webview initialization Error: Failed to find path to launch Storybook server
Error: Failed to find path to launch Storybook server
    at LT.start (/Users/ianvs/.vscode/extensions/joshbolduc.story-explorer-0.5.8/dist/extension/index.js:2445:4037)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at /Users/ianvs/.vscode/extensions/joshbolduc.story-explorer-0.5.8/dist/extension/index.js:2446:1088
```

**Comment by joshbolduc** [comment]
That pretty much confirms the issue is here:
https://github.com/joshbolduc/vscode-story-explorer/blob/f9ed1bf64cf73941d1f04a3ae11ba9d55a94b655/src/server/getStorybookBinPath.ts#L11-L14

but I don't have any good theories as to why that wouldn't return any results for you, given that specifying the path manually works and I can't reproduce the issue at all.

I did find a vaguely similar sounding issue in vscode: https://github.com/microsoft/vscode/issues/147235

I don't know if it's related, but if you're up for it, it sounds like [there's a way to get some extra logs](https://github.com/microsoft/vscode/issues/147235#issuecomment-1097298192) from the VS Code side. Maybe that would reveal something interesting?

**Comment by IanVS** [comment]
Thanks, Here's what I see when I turn on trace logs:

```
[2022-07-05 21:33:26.633] [renderer5] [trace] CommandService#executeCommand _vscode_delegate_cmd_l58xcjxv
[2022-07-05 21:33:26.647] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.662] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.662] [renderer5] [trace] SearchService#search {"_reason":"startFileSearch","folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/ianvs/code/defined/webclient","external":"file:///Users/ianvs/code/defined/webclient","path":"/Users/ianvs/code/defined/webclient","scheme":"file"},"fileEncoding":"utf8","disregardIgnoreFiles":true,"disregardGlobalIgnoreFiles":true,"disregardParentIgnoreFiles":true,"ignoreSymlinks":true}],"usingSearchPaths":false,"includePattern":{"**/node_modules/.bin/start-storybook":true},"type":1}
[2022-07-05 21:33:26.663] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.734] [renderer5] [trace] [History stack global-default]: notifyNavigation() (editor: webview-panel:webview-panel/webview-3fcc4e33-141b-4aa5-b106-54b6abcedfa9, event: <none>)
[2022-07-05 21:33:26.735] [renderer5] [trace] [History stack global-default]: notifyNavigation() not ignoring (editor: webview-panel:webview-panel/webview-3fcc4e33-141b-4aa5-b106-54b6abcedfa9, event: <none>)
[2022-07-05 21:33:26.735] [renderer5] [trace] [History stack global-default]: onNonSelectionAwareEditorNavigation() (editor: webview-panel:webview-panel/webview-3fcc4e33-141b-4aa5-b106-54b6abcedfa9, event: <none>)
[2022-07-05 21:33:26.735] [renderer5] [trace] [History stack global-default]: add() (editor: webview-panel:webview-panel/webview-3fcc4e33-141b-4aa5-b106-54b6abcedfa9, event: <none>)
[2022-07-05 21:33:26.736] [renderer5] [trace] [History stack global-default]: index: 7, navigating: false
- group: 0, editor: file:///Users/ianvs/code/defined/webclient/eslint-rules/no-multiple-use-url-param.test.js, selection: line: 30-30, col:  43-43
- group: 0, editor: file:///Users/ianvs/code/defined/webclient/eslint-rules/no-multiple-use-url-param.js, selection: line: 1-1, col:  4-4
- group: 0, editor: file:///Users/ianvs/code/defined/webclient/eslint-rules/package.json, selection: line: 4-4, col:  1-1
- group: 0, editor: file:///Users/ianvs/code/defined/webclient/eslint-rules/no-multiple-use-url-param.js, selection: line: 2-2, col:  1-1
- group: 0, editor: file:///Users/ianvs/code/defined/webclient/eslint-rules/no-multiple-use-url-param.test.js, selection: line: 30-30, col:  43-43
- group: 0, editor: file:///Users/ianvs/code/defined/webclient/src/screens/hosts/hosts-page/HostsPage.tsx, selection: line: 23-23, col:  53-53
- group: 0, editor: file:///Users/ianvs/code/defined/webclient/src/screens/errors/OopsPage.stories.tsx, selection: line: 1-1, col:  1-1
- group: 1, editor: webview-panel:webview-panel/webview-3fcc4e33-141b-4aa5-b106-54b6abcedfa9, selection: <none>
			
[2022-07-05 21:33:26.748] [renderer5] [debug] Webview(3fcc4e33-141b-4aa5-b106-54b6abcedfa9): will update content
[2022-07-05 21:33:26.803] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.804] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.804] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.805] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.805] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.805] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.843] [renderer5] [trace] [DEBOUNCE: InlayHint] for file:///Users/ianvs/code/defined/webclient/src/screens/errors/OopsPage.stories.tsx is 40ms
[2022-07-05 21:33:26.850] [renderer5] [trace] MainThreadTreeViews#$refresh storyExplorer.storiesView undefined
[2022-07-05 21:33:26.851] [renderer5] [trace] [DEBOUNCE: DocumentRangeSemanticTokens] for file:///Users/ianvs/code/defined/webclient/src/screens/errors/OopsPage.stories.tsx is 100ms
[2022-07-05 21:33:26.851] [renderer5] [trace] SemanticTokensProviderStyling [CACHED] 7 / 9: foreground 18, fontStyle 0
... bunch more like ^
[2022-07-05 21:33:26.914] [renderer5] [debug] Webview(3fcc4e33-141b-4aa5-b106-54b6abcedfa9): webview ready
[2022-07-05 21:33:26.920] [renderer5] [trace] CommandService#executeCommand _executeCodeLensProvider
[2022-07-05 21:33:26.939] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.951] [renderer5] [trace] [DEBOUNCE: CodeLensProvide] for file:///Users/ianvs/code/defined/webclient/src/screens/errors/OopsPage.stories.tsx is 250ms
[2022-07-05 21:33:26.965] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.968] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.968] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.969] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.969] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:26.970] [renderer5] [trace] [DEBOUNCE: CodeLensProvide] for file:///Users/ianvs/code/defined/webclient/src/screens/errors/OopsPage.stories.tsx is 250ms
[2022-07-05 21:33:27.022] [renderer5] [trace] SearchService#search: 218ms
[2022-07-05 21:33:27.026] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:27.181] [renderer5] [trace] CommandService#executeCommand setContext
[2022-07-05 21:33:27.181] [renderer5] [trace] MainThreadTreeViews#$refresh storyExplorer.storiesView undefined
[2022-07-05 21:33:27.182] [renderer5] [trace] CommandService#executeCommand _executeCodeLensProvider
[2022-07-05 21:33:27.182] [renderer5] [error] [Extension Host] Failed to complete webview initialization Error: Failed to find path to launch Storybook server
    at LT.start (/Users/ianvs/.vscode/extensions/joshbolduc.story-explorer-0.5.8/dist/extension/index.js:2445:4037)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at /Users/ianvs/.vscode/extensions/joshbolduc.story-explorer-0.5.8/dist/extension/index.js:2446:1088
```

Interestingly, I also notice that in a steady state, with nothing else happening, I get these pairs of logs about once every 0.7 seconds:

```
[2022-07-05 21:35:57.188] [renderer5] [trace] CommandService#executeCommand _executeCodeLensProvider
[2022-07-05 21:35:57.445] [renderer5] [trace] MainThreadTreeViews#$refresh storyExplorer.storiesView undefined
[2022-07-05 21:35:57.685] [renderer5] [trace] MainThreadTreeViews#$refresh storyExplorer.storiesView undefined
```

I'll try cloning the repo and play around with the code a bit and see if I can figure out what's going on.

**Comment by IanVS** [comment]
Alright, I finally figured it out.  I had `"search.followSymlinks": false` in my settings, and `workspace.findFiles` uses that setting as well. I still can't get the preview to load yet, but at least this part of the mystery is solved.

**Comment by joshbolduc** [comment]
Aha! Good find. I tried copying your settings, but I must've overlooked that `storyExplorer.server.internal.storybookBinaryPath` was included in there too, so I had ruled it out. My bad.

I don't think it's going to be intuitive to folks that this setting affects the auto-launch behavior. I'll need to see how I can address it for others who also disable that setting.

And thanks for the heads up about the log noise--I just took care of that in #473.
